lib/web_ui/lib/src/engine/canvaskit/embedded_views.dart

```diff
@@ -246,11 +246,26 @@ class HtmlViewEmbedder {
     return head;
   }
 
-  void _applyMutators(MutatorsStack mutators, html.Element embeddedView) {
+  /// Clean up the old SVG clip definitions, as this platform view is about to
+  /// be recomposited.
+  void _cleanUpClipDefs(int viewId) {
+    if (_svgClipDefs.containsKey(viewId)) {
+      final html.Element clipDefs =
+          _svgPathDefs!.querySelector('#sk_path_defs')!;
+      for (String id in _svgClipDefs[viewId]!) {
+        clipDefs.querySelector('#$id')?.remove();
```

Comment:
Should we keep a reference to the actual DOM Node of each clipDef (`newClipPath`?), so we don't have to do any DOM lookups here?

Reply:
That's how I tried to do it at first (keeping a `Map<int, List<html.Node>>`... but for whatever reason, when I remove the `html.Node` created by `html.DocumentFragment.svg`, it doesn't remove the `<clipPath>` element from the DOM...

Reply:
I added a comment explaining why we can't just store the node directly.

Reply:
`querySelector` used to be extremely expensive. I think the issue is that DocumentFragment is a transient concept. When attaching it to the DOM it simply iterates over its children and attaches them one by one. The fragment itself is not attached. Would it be possible to wrap all clip elements under a common parent and then attach the parent? This way we can keep the reference to the parent.

Reply:
The common parent exists, it seems to be the `clipDef` tag. Maybe we can just inject it into the page just in case it's needed (right now, it's being probed every time there's a mutator of type clip, it seems)